src/API/CacheAwareService.php

```diff
@@ -0,0 +1,323 @@
+<?php
+	/*
+	 * Copyright (c) 2017 Kruithne (kruithne@gmail.com)
+	 * https://github.com/Kruithne/KrameWork7
+	 *
+	 * Permission is hereby granted, free of charge, to any person obtaining a copy
+	 * of this software and associated documentation files (the "Software"), to deal
+	 * in the Software without restriction, including without limitation the rights
+	 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+	 * copies of the Software, and to permit persons to whom the Software is
+	 * furnished to do so, subject to the following conditions:
+	 *
+	 * The above copyright notice and this permission notice shall be included in all
+	 * copies or substantial portions of the Software.
+	 *
+	 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+	 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+	 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+	 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+	 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+	 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+	 * SOFTWARE.
+	 */
+
+	namespace KrameWork\API;
+
+	/**
+	 * Class ServiceResponse
+	 * A service call response to encode as JSON and return to the caller
+	 *
+	 * @package KrameWork\API
+	 * @author docpify <morten@runsafe.no>
+	 */
+	class ServiceResponse implements \JsonSerializable
+	{
+		/**
+		 * ServiceResponse constructor.
+		 *
+		 * @api __construct
+		 * @param bool $success
+		 * @param mixed $data
+		 */
+		public function __construct(bool $success = true, $data = []) {

```

Comment:
$data should be type-hinted to array as its used as an array later on, unless its an object that implements \ArrayAccess